# Homework #3
|Author| Stanley A. Baronett|
|--|-------------------------------|
|Created | 9/22/2020|
|Updated | 9/24/2020|

All references to chapter problems, numbered equations, and page numbers are from _Astrophysics in a Nutshell: Second Edition_ by Dan Maoz.

## Chapter 3, Problem 2

From Eq. 3.76, 

\begin{align}
P = P_\textrm{g} + P_\textrm{rad} = \dfrac{\rho kT}{\bar m} + \dfrac{1}{3}aT^4,
\tag{A}
\end{align}

we can separately define the pressure inside a normal star, due to _gas_ ($P_\textrm{g}$) and _radiation_ ($P_\textrm{rad}$), as follows:

In [23]:
def gas_press(rho, T, mbar):
    """
    The gas pressure equation of state (Eq. 3.69), where 'rho' is the density,
    'k' is Boltzmann's constant, 'T' is the temperature, and 'mbar' is the mean
    particle mass.
    """
    k = 1.4e-16         # erg/K
    return rho*k*T/mbar # in cgs units barye (Ba = dyn/cm^2)

def rad_press(T):
    """
    The radiation pressure equation of state (Eq. 3.75), where 'a' is the
    radiation constant, and 'T' is the temperature.
    """
    a = 7.6e-15     # radiation constant in erg/cm^3/K^4
    return a*T**4/3 # in cgs units barye (Ba = dyn/cm^2)

Using appropriate solar parameters, e.g., mean density ($\bar \rho$), _virial_ temperature ($T_\textrm{vir}$), and mean particle mass ($\bar m$), we can calculate the typical gas and radiation pressures throughout the Sun. "For solar abundances, $X = 0.71, Y=0.27, Z=0.02$, and therefore $\bar m/m_\textrm{H} = 0.61$." (p. 43)

In [21]:
# Sun's general parameters
rhobar = 1.4       # g/cm^3 (Eq. 3.4)
Tvir   = 4e6       # virial temp. in K (p. 35)
m_H    = 1.7e-24   # mass of H atom in g
mbar   = 0.61*m_H  # mean particle mass in g (p. 43)

Pg = gas_press(rhobar, Tvir, mbar)
Prad = rad_press(Tvir)

print('Throughout the Sun (on average):')
print('Pg      = {:.2e} Ba'.format(Pg))
print('Prad    = {:.2e} Ba'.format(Prad))
print('Pg/Prad = {:.2e}'.format(Pg/Prad))

Throughout the Sun (on average):
Pg      = 7.56e+14 Ba
Prad    = 6.49e+11 Ba
Pg/Prad = 1.17e+03


Thus, the __kinetic pressure dominates__ that from radiation __by 3 orders of magnitude__.

Similarly, using the central density ($\rho_\textrm{c}$) and temperature ($T_\textrm{c}$) of the Sun, we can calculate the gas and radiation pressures in the core. "In the central regions of the Sun, about half of the hydrogen has already been converted into helium by nuclear reactions, and as a result $X = 0.34, Y=0.64, Z=0.02$, giving $\bar m/m_\textrm{H} = 0.85$." (p. 43)

In [24]:
# Sun's core parameters
rho_c  = 150      # g/cm^3 (Eq. 3.10)
Tc     = 1.5e6    # K      (Eq. 3.11)
mbar_c = 0.85*m_H # mean particle mass in g (p. 43)

Pg = gas_press(rho_c, Tc, mbar_c)
Prad = rad_press(Tc)

print('In the Sun\'s core (central regions):')
print('Pg      = {:.2e} Ba'.format(Pg))
print('Prad    = {:.2e} Ba'.format(Prad))
print('Pg/Prad = {:.2e}'.format(Pg/Prad))

In the Sun's core (central regions):
Pg      = 2.18e+16 Ba
Prad    = 1.28e+10 Ba
Pg/Prad = 1.70e+06


Even in the core, where the internal temperature is about $10^7$ K, the kinetic pressure dominates that from radiation, this time by __6 orders of magnitude__.

## Chapter 3, Problem 4

Consider a hypothetical star of radius $R$, with density $\rho$ that is constant, i.e., independent of radius. The star is composed of a classical, nonrelativistic, ideal gas of fully ionized hydrogen.

### a.
From Eq. 3.56, the differential equation of stellar structure for pressure is

\begin{align}
\dfrac{dP(r)}{dr} = -\dfrac{GM(r)\rho}{r^2}
\tag{B}
\end{align}

Solving for the pressure profile, $P(r)$,

\begin{align}
\int_r^RdP(r) &= -\int_r^R\dfrac{GM(r)\rho}{r^2}dr \\
P(R) - P(r) &= -G\rho\int_r^R\dfrac{M(r)}{r^2}dr.
\end{align}

Applying the boundary condition $P(R) = 0$,

\begin{align}
P(r) &= G\rho\int_r^R\dfrac{M(r)}{r^2}dr.
\tag{C}
\end{align}

Since $\rho$ is constant,

\begin{align}
M(r) = \rho V(r) = \rho\dfrac{4}{3}\pi r^3.
\end{align}

Substituting into Eq. C,

\begin{align}
P(r) &= G\rho\int_r^R\dfrac{4\rho\pi r^3}{3r^2}dr \\
     &= \dfrac{4}{3}\pi G\rho^2 \int_r^R rdr \\
     &= \dfrac{2\pi G\rho^2}{3}\left[r^2\right|_r^R \\
    &= \boxed{\dfrac{2\pi}{3}G\rho^2(R^2 - r^2)}
\tag{D}
\end{align}

### b.

From Eq. 3.69, the gas pressure from the equation of state of a classical, nonrelativistic, ideal gas, is

\begin{align}
P_g = \dfrac{\rho}{\bar m}kT,
\tag{E}
\end{align}

where, from Eq. 3.70, $\bar m = m_\textrm{H}/2$ is the mean particle mass for completely ionized pure hydrogen. Setting this equal to our expression from part __a__ (Eq. D) and solving for $T$, we find the temperature profile as

\begin{align}
\dfrac{2\rho}{m_\textrm{H}}kT(r) &= \dfrac{2\pi}{3}G\rho^2(R^2 - r^2) \\
T(r) &= \boxed{\dfrac{\pi}{3k}Gm_\textrm{H}\rho(R^2 - r^2)}.
\tag{F}
\end{align}

### c.

If we assume the nuclear energy production rate depends on temperature as $\epsilon \sim T^4$, then

\begin{align}
\epsilon \sim (R^2 - r^2)^4,
\tag{G}
\end{align}

where we've ignored the constants of proportionality in Eq. F. It's central value, at $r = 0$, is

\begin{align}
\epsilon_\textrm{c} &\sim R^8
\tag{H}
\end{align}

The radius at which $\epsilon$ decreases to $0.1\epsilon_\textrm{c}$ is

\begin{align}
0.1\epsilon_\textrm{c} &\sim (R^2 - r^2)^4 \\
0.1R^8 &\sim R^8 \left( 1 - \dfrac{r^2}{R^2} \right) \\
(0.1)^{1/4} &\sim 1 - \dfrac{r^2}{R^2} \\
r &\sim \left( 1 - (0.1)^{1/4} \right)^{1/2}R.
\tag{I}
\end{align}

In [25]:
(1 - 0.1**(1/4))**(1/2)

0.6615577637739964

\begin{equation}
\boxed{r(\epsilon = 0.1\epsilon_\textrm{c}) \sim 0.66R}
\tag{J}
\end{equation}

In [26]:
0.66**3

0.28749600000000003

Since the volume of a sphere is $V = \frac{3}{4}\pi r^3$, the fraction of the star's total volume included within this radius is

\begin{equation}
\boxed{0.66^3 \approx 29\%}
\tag{K}
\end{equation}

## Chapter 3, Problem 6

Consider a star of mass $M = 10M_\odot$, composed entirely of fully ionized $^{12}\textrm{C}$. Its core temperature is $T_\textrm{c} = 6 \times 10^8 \,\textrm{K}$ (compared to $T_{\textrm{c},\odot} = 1.5 \times 10^7 \,\textrm{K}$ for the Sun).

### a.

